rskj-core/src/main/java/org/ethereum/facade/EthereumImpl.java

```diff
@@ -124,7 +107,6 @@ public void connect(InetAddress addr, int port, String remoteId) {
     @Override
     public void connect(final String ip, final int port, final String remoteId) {
         logger.info("Connecting to: {}:{}", ip, port);
-        final PeerClient peerClient = ctx.getBean(PeerClient.class);
```

Comment:
I'm worried about the lifespan here. Doesn't this instance a new `PeerClient` every time? But if we inject it via constructor, then we will always have the same `peerClient` instance.
I'm invoking @ajlopez here to check.

Reply:
Another great catch, I didn't realize the `@Scope("prototype")` of the **PeerClient**. I'll take another look tomorrow.